src/Builder/Creator/MethodCreator.php

```diff
@@ -0,0 +1,233 @@
+            ['flags' => Class_::MODIFIER_PUBLIC, 'returnType' => $returnType],
+            ['comments' => [$this->methodDocBlockCreator->createMethodDocBlock($value)]],
+        );
+        $nodeTraverser->addVisitor(new AddMethodVisitor($classMethod));
+
+        return $classInformationTransfer
+            ->setClassTokenTree($nodeTraverser->traverse($classInformationTransfer->getClassTokenTree()));
+    }
+
+    /**
+     * @return \PhpParser\Parser
+     */
+    protected function createParser(): Parser
+    {
+        return (new ParserFactory())->create(ParserFactory::PREFER_PHP7);
+    }
+
+    /**
+     * @param \SprykerSdk\Integrator\Transfer\ClassInformationTransfer $classInformationTransfer
+     * @param mixed $value
+     *
+     * @return array<\PhpParser\Node\Stmt\Return_>
+     */
+    protected function createReturnArrayStatement(ClassInformationTransfer $classInformationTransfer, $value): array
+    {
+        $tree = $this->methodStatementsCreator->createMethodStatementsFromValue($classInformationTransfer, $value);
+
+        return [new Return_(new Array_($tree))];
+    }
+
+    /**
+     * @param array<\PhpParser\Node\Stmt> $tree
+     *
+     * @return bool
+     */
+    protected function isSingleReturnStatement(array $tree): bool
+    {
+        return $tree && count($tree) === static::SINGLE_TREE_RETURN_STATEMENT_COUNT_ELEMENTS;
+    }
+
+    /**
+     * @param array<\PhpParser\Node\Stmt> $tree
+     *
+     * @return array
+     */
+    protected function createMultiStatementMethodBody(array $tree): array
+    {
+        $lastElement = $tree[count($tree) - 1];
+        if (!$lastElement instanceof Return_ && property_exists($lastElement, 'expr')) {
+            $tree[count($tree) - 1] = new Return_($lastElement->expr);
+        }
+
+        return $tree;
+    }
+
+    /**
+     * @param \SprykerSdk\Integrator\Transfer\ClassInformationTransfer $classInformationTransfer
+     * @param array $tree
+     * @param mixed $value
+     *
+     * @throws \SprykerSdk\Integrator\Builder\Exception\LiteralValueParsingException
+     *
+     * @return array
+     */
+    protected function createSingleStatementMethodBody(ClassInformationTransfer $classInformationTransfer, array $tree, $value): array
+    {
+        /** @var \PhpParser\Node\Stmt\Expression|null $returnExpression */
+        $returnExpression = $tree[0] ?? null;
+        if (!$returnExpression) {
+            throw new LiteralValueParsingException(sprintf('Value is not valid PHP code: `%s`', $value));
+        }
+        if (property_exists($returnExpression->expr, 'class')) {
+            return $this->createClassConstantReturnStatement($classInformationTransfer, $returnExpression);
+        }
+
+        return [new Return_($returnExpression->expr)];
+    }
+
+    /**
+     * @param \SprykerSdk\Integrator\Transfer\ClassInformationTransfer $classInformationTransfer
+     * @param \PhpParser\Node\Stmt\Expression $expression
+     *
+     * @return array
```

Comment:
not a blocker, but would be nice to have smth like array<\PhpParser\Node\Stmt\Return_> here